src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -5048,34 +5049,141 @@ namespace winrt::TerminalApp::implementation
+        MUX::Controls::CommandBarFlyout splitPaneRightMenu{};
+        MUX::Controls::CommandBarFlyout splitPaneLeftMenu{};
+
         // Wire up each item to the action that should be performed. By actually
         // connecting these to actions, we ensure the implementation is
         // consistent. This also leaves room for customizing this menu with
         // actions in the future.
 
-        makeItem(RS_(L"SplitPaneText"), L"\xF246", ActionAndArgs{ ShortcutAction::SplitPane, SplitPaneArgs{ SplitType::Duplicate } });
-        makeItem(RS_(L"DuplicateTabText"), L"\xF5ED", ActionAndArgs{ ShortcutAction::DuplicateTab, nullptr });
+        if (!withSelection)
+        {
+            makeItem(RS_(L"FindText"), L"\xF78B", ActionAndArgs{ ShortcutAction::Find, nullptr }, menu);
+        }
+
+        makeItem(RS_(L"DuplicateTabText"), L"\xF5ED", ActionAndArgs{ ShortcutAction::DuplicateTab, nullptr }, menu);
+
+        makeItem(RS_(L"SplitPaneDuplicateText") + L" " + focusedProfile.Name(), focusedProfile.Icon(), ActionAndArgs{ ShortcutAction::SplitPane, SplitPaneArgs{ SplitType::Duplicate, SplitDirection::Down, .5, nullptr } }, splitPaneDownMenu);
+        makeItem(RS_(L"SplitPaneDuplicateText") + L" " + focusedProfile.Name(), focusedProfile.Icon(), ActionAndArgs{ ShortcutAction::SplitPane, SplitPaneArgs{ SplitType::Duplicate, SplitDirection::Up, .5, nullptr } }, splitPaneUpMenu);
+        makeItem(RS_(L"SplitPaneDuplicateText") + L" " + focusedProfile.Name(), focusedProfile.Icon(), ActionAndArgs{ ShortcutAction::SplitPane, SplitPaneArgs{ SplitType::Duplicate, SplitDirection::Right, .5, nullptr } }, splitPaneRightMenu);
+        makeItem(RS_(L"SplitPaneDuplicateText") + L" " + focusedProfile.Name(), focusedProfile.Icon(), ActionAndArgs{ ShortcutAction::SplitPane, SplitPaneArgs{ SplitType::Duplicate, SplitDirection::Left, .5, nullptr } }, splitPaneLeftMenu);
+
+        // add menu separator
+        const auto separatorDownItem = AppBarSeparator{};
+        const auto separatorUpItem = AppBarSeparator{};
+        const auto separatorRightItem = AppBarSeparator{};
+        const auto separatorLeftItem = AppBarSeparator{};
+
+        splitPaneDownMenu.SecondaryCommands().Append(separatorDownItem);
+        splitPaneUpMenu.SecondaryCommands().Append(separatorUpItem);
+        splitPaneRightMenu.SecondaryCommands().Append(separatorRightItem);
+        splitPaneLeftMenu.SecondaryCommands().Append(separatorLeftItem);
+
+        for (auto profileIndex = 0; profileIndex < activeProfileCount; profileIndex++)
+        {
+            const auto profile = activeProfiles.GetAt(profileIndex);
+            NewTerminalArgs args{};
+            args.Profile(profile.Name());
+            args.StartingDirectory(_evaluatePathForCwd(profile.EvaluatedStartingDirectory()));
+            args.TabTitle(profile.TabTitle());
+            args.Commandline(profile.Commandline());
+            args.SuppressApplicationTitle(profile.SuppressApplicationTitle());
```

Comment:
I think all these properties _should_ just be implicit if you pass the `Profile` in. No need to manually override the value from the profile, _with the value from the profile_, right?

Reply:
```
for (auto profileIndex = 0; profileIndex < activeProfileCount; profileIndex++)
{
    const auto profile = activeProfiles.GetAt(profileIndex);
    NewTerminalArgs args{};
    args.Profile(profile.Name());
    args.StartingDirectory(_evaluatePathForCwd(profile.EvaluatedStartingDirectory()));
    args.TabTitle(profile.TabTitle());
    args.Commandline(profile.Commandline());
    args.SuppressApplicationTitle(profile.SuppressApplicationTitle());
    makeItem(profile.Name(), profile.Icon(), ActionAndArgs{ ShortcutAction::SplitPane, SplitPaneArgs{ SplitType::Manual, SplitDirection::Down, .5, args } }, splitPaneDownMenu);
    makeItem(profile.Name(), profile.Icon(), ActionAndArgs{ ShortcutAction::SplitPane, SplitPaneArgs{ SplitType::Manual, SplitDirection::Up, .5, args } }, splitPaneUpMenu);
    makeItem(profile.Name(), profile.Icon(), ActionAndArgs{ ShortcutAction::SplitPane, SplitPaneArgs{ SplitType::Manual, SplitDirection::Right, .5, args } }, splitPaneRightMenu);
    makeItem(profile.Name(), profile.Icon(), ActionAndArgs{ ShortcutAction::SplitPane, SplitPaneArgs{ SplitType::Manual, SplitDirection::Left, .5, args } }, splitPaneLeftMenu);
}
```
You mean add overloaded function for:
args.Profile
that will get Profile type as argument and not profile name ?
like:
args.Profile(profile.Name());